x-pack/legacy/plugins/file_upload/public/components/json_index_file_picker.js

```diff
@@ -21,117 +21,114 @@ export class JsonIndexFilePicker extends Component {
   state = {
     fileUploadError: '',
     fileParsingProgress: '',
-    fileRef: null
   };
 
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.fileRef !== this.props.fileRef) {
-      this.setState({ fileRef: this.props.fileRef });
-    }
-  }
-
   _fileHandler = async fileList => {
-    const {
-      resetFileAndIndexSettings, setParsedFile, onFileRemove, onFileUpload,
-      transformDetails, setFileRef, setIndexName
-    } = this.props;
-
-    const { fileRef } = this.state;
+    const fileListArr = Array.from(fileList);
+    await this._validateAndParse(fileListArr);
+  };
 
-    resetFileAndIndexSettings();
+  _validateAndParse = async fileArr => {
+    this.props.resetFileAndIndexSettings();
     this.setState({ fileUploadError: '' });
-    if (fileList.length === 0) { // Remove
-      setParsedFile(null);
-      if (onFileRemove) {
-        onFileRemove(fileRef);
-      }
-    } else if (fileList.length === 1) { // Parse & index file
-      const file = fileList[0];
-      if (!file.name) {
-        this.setState({
-          fileUploadError: i18n.translate(
-            'xpack.fileUpload.jsonIndexFilePicker.noFileNameError',
-            { defaultMessage: 'No file name provided' })
-        });
-        return;
-      }
-
-      // Check file type, assign default index name
-      const splitNameArr = file.name.split('.');
-      const fileType = splitNameArr.pop();
-      const types = ACCEPTABLE_FILETYPES.reduce((accu, type) => {
-        accu = accu ? `${accu}, ${type}` : type;
-        return accu;
-      }, '');
-      if (!ACCEPTABLE_FILETYPES.includes(fileType)) {
-        this.setState({
-          fileUploadError: (
-            <FormattedMessage
-              id="xpack.fileUpload.jsonIndexFilePicker.acceptableTypesError"
-              defaultMessage="File is not one of acceptable types: {types}"
-              values={{ types }}
-            />
-          )
-        });
+    if (fileArr.length === 0) { // Remove
+      return;
+    } else if (fileArr.length === 1) { // Parse & index file
```

Comment:
No need for if else block since first if returns. Just make this an if statement then you don't need the empty else block later